python/ray/serve/_private/deployment_state.py

```diff
@@ -2004,39 +2004,32 @@ def _get_all_node_ids(self):
         # Test mock purpose
         return get_all_node_ids(self._gcs_client)
 
-    def _deploy_driver(self) -> bool:
+    def _deploy_driver(self) -> List[ReplicaSchedulingRequest]:
         """Deploy the driver deployment to each node."""
-        all_nodes = self._get_all_node_ids()
-        deployed_nodes = set()
-        for replica in self._replicas.get(
-            [
-                ReplicaState.STARTING,
-                ReplicaState.RUNNING,
-                ReplicaState.RECOVERING,
-                ReplicaState.UPDATING,
-                ReplicaState.STOPPING,
-            ]
-        ):
-            if replica.actor_node_id:
-                deployed_nodes.add(replica.actor_node_id)
-        replica_changed = False
-        for node_id, _ in all_nodes:
-            if node_id in deployed_nodes:
-                continue
+        upscale = []
+        num_existing_replicas = self._replicas.count()
+        if num_existing_replicas >= self._target_state.num_replicas:
+            num_running_replicas = self._replicas.count(states=[ReplicaState.RUNNING])
+            if num_running_replicas >= self._target_state.num_replicas:
+                for replica in self._replicas.pop(states=[ReplicaState.STARTING]):
+                    self._stop_replica(replica)
+
+            return upscale
```

Comment:
confused by this return -- why are we early returning `upscale` list in the downscale case?

Reply:
is this just to early return with an empty list?

Reply:
Yea, just early return an empty list. I changed to return an empty list directly to be obvious